src/Compilers/CSharp/Portable/Binder/WithLambdaParametersBinder.cs

```diff
@@ -135,13 +135,6 @@ private bool ReportConflictWithParameter(ParameterSymbol parameter, Symbol newSy
                 return true;
             }
 
-            if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local)
```

Comment:
Expand the review up a few lines and you'll see an identical check being made which unconditionally returns in the `if` body.  Hence this code path should never get hit.